sklearn/pipeline.py

```diff
@@ -217,122 +326,144 @@ def fit_predict(self, X, y=None, **fit_params):
         X : iterable
             Training data. Must fulfill input requirements of first step of
             the pipeline.
+
         y : iterable, default=None
             Training targets. Must fulfill label requirements for all steps
             of the pipeline.
+
+        fit_params : dict of string -> object
+            Parameters passed to the `fit` method of each step, where
+            each parameter name is prefixed such that parameter ``p`` for step
+            ``s`` has key ``s__p``.
         """
-        Xt, fit_params = self._pre_transform(X, y, **fit_params)
+        Xt = X
+        for name, transform in self.steps[:-1]:
+            if transform is not None:
```

Comment:
alternative: if someone sets something to ``None`` replace it with ``FunctionTransformer(lambda x: x)`` .... not sure if that is actually easier...